packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2253,8 +2262,25 @@ class InputDecoration {
   /// null.
   ///
   /// The semantic label can be replaced by providing a [semanticCounterText].
+  ///
+  /// If set to empty string and [counter] isn't specified, will display nothing
+  /// in the place of the counter.
   final String counterText;
 
+  /// Optional function to generate a custom counter widget, called with
+  /// positional arguments [int currentLength] and [int maxLength].  Will be
+  /// placed below the line instead of the default widget built when passing
+  /// [counterText].
+  ///
+  /// {@tool sample}
+  /// ```dart
+  /// Widget counter(int currentLength, int maxLength) {
```

Comment:
This example is clear but someone trying it out might discover that the counter text doesn't really fit. Might be better to use a shorter version.